File: StayBookingFrontend/lib/model/create_room_request_dto.dart
import 'package:stay_booking_frontend/model/room_upload_file.dart';

class CreateRoomRequestDto {
  CreateRoomRequestDto({
    required this.hotelId,
    required this.roomType,
    required this.description,
    required this.price,
    required this.available,
    required this.createdBy,
    required this.photoFiles,
  });

  final int hotelId;
  final String roomType;
  final String description;
  final double price;
  final bool available;
  final String createdBy;
  final List<RoomUploadFile> photoFiles;
}

File: StayBooking/StayBooking/src/main/java/com/dto/CreateRoomRequestDto.java
package com.dto;

import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.math.BigDecimal;
import java.util.List;

@Data
public class CreateRoomRequestDto {
    private Integer hotelId;
    private String hotelName;
    private String roomType;
    private String description;
    private BigDecimal price;
    private Boolean available;
    private String createdby;
    private List<MultipartFile> photos;
}
